src/v/config/configuration.cc

```diff
@@ -2506,6 +2506,28 @@ configuration::configuration()
       "Size of the zstd decompression workspace",
       {.visibility = visibility::tunable},
       8_MiB)
+  , lz4_decompress_buffer_size_bytes(
+      *this,
+      "lz4_decompress_buffer_size_bytes",
+      "Size of buffers preallocated for LZ4 decompression. Two buffers "
+      "allocated per shard, so twice of this value is reserved at startup",
+      {.needs_restart = needs_restart::yes, .visibility = visibility::tunable},
+      4_MiB + 128_KiB)
+  , lz4_decompress_malloc_min_threshold(
+      *this,
+      "lz4_decompress_malloc_min_threshold",
+      "Threshold below which preallocated buffers will not be used during "
+      "LZ4 decompression, if preallocated buffer use is enabled",
+      {.needs_restart = needs_restart::no, .visibility = visibility::tunable},
+      128_KiB + 1)
```

Comment:
is there some advantage to not using pre-allocated buffers? if they are there anyway, why not always use them?